modules/ray_tracing/doc/content/source/userobject/RayTracingStudy.md

```diff
@@ -0,0 +1,97 @@
+
+The enabling and disabling of ray registration is handled by the `_use_ray_registration` private parameter, for which the default is true. When ray registration is enabled, all [RayKernels/index.md] and [RayBCs/index.md] +must+ have the parameter [!param](/RayKernels/NullRayKernel/rays) set, which identifies the [Rays](Ray.md) by name that said objects are executed on. When ray registration is disabled, the [!param](/RayKernels/NullRayKernel/rays) parameter cannot be used and [RayKernels/index.md] and [RayBCs/index.md] will be executed for +all+ [Rays](Ray.md).
+
+Use the protected methods `registerRay()` and `registerRays()` for the registration of [Ray(s)](Ray.md) with the given name(s). These methods will return the ID(s) that are to be assigned to the [Ray(s)](Ray.md). In order to use ray registration, the registration methods must be called on all processors at the same time.
+
+The public methods `registeredRayID()` and `registeredRayName()` can be utilized to map a registered [Ray.md] name to its ID and an ID to its registered name, respectively.
+
+### Ray Banking
+
+It is often useful to examine a [Ray.md] after is has completed tracing. An example use case is for obtaining the accumulated integral for a [Ray.md] that is integrating a field along a line (see [line_integrals.md] and [IntegralRayKernel.md]). Within line integrals, the [Ray.md] banks will be accessed to obtain the final accumulated value in a [Ray.md].
+
+With the `_bank_rays_on_completion` private parameter set to true, all [Rays](Ray.md) that complete on a given processor are stored in the private member variable `_ray_bank`. This bank can be accessed through the `rayBank()` method or via `getBankedRay()`, which will return the [Ray.md] with the requested ID on the processor that ended the [Ray.md].
+
+In addition, the methods `getBankedRayData()` and `getBankedRayAuxData()` are available to get a single data value from a [Ray.md] from the banks after it has completed. For these methods, the resulting value is replicated across all processors.
+
+### Ray Pool
+
+For simulations with the [ray_tracing/index.md] that generate of a large number of [Ray.md] objects, it becomes advantageous to minimize the construction of new [Ray.md] objects. More so whenever the size of the data on the [Ray.md] is significant.
+
+[Ray.md] storage is typically wrapped as a `std::shared_ptr<Ray>`. The significance of using a `shared_ptr` is that the destructor is called when the use count reaches zero, that is, when nobody is holding onto the [Ray.md]. The internal tracing algorithm will decrease the use count for a traced [Ray.md] after it has completed tracing. That is, if no other shared ownership of the [Ray.md] exists, the use count will reach zero. This is why [#ray-banking] exists - the [Ray.md] bank will increase the use count of the [Ray.md] so that it is available for use after tracing and is not destructed.
```

Comment:
```suggestion
[Ray.md] storage is typically wrapped as a `std::shared_ptr<Ray>`. The significance of using a `shared_ptr` is that the destructor is called when the use count reaches zero, that is, when nobody is holding onto the [Ray.md]. The internal tracing algorithm will decrease the use count for a traced [Ray.md] after it has completed tracing. That is, if no other shared ownership of the [Ray.md] exists, the use count will reach zero. This is why [#ray-banking] exists---the [Ray.md] bank will increase the use count of the [Ray.md] so that it is available for use after tracing and is not destructed.
```